sdk/core/azure-core/azure/core/pipeline/policies/_authentication_async.py

```diff
@@ -164,4 +164,9 @@ def on_exception(self, request: PipelineRequest[HTTPRequestType]) -> None:
         return
 
     def _need_new_token(self) -> bool:
-        return not self._token or self._token.expires_on - time.time() < 300
+        now = time.time()
+        return (
+            not self._token
+            or (self._token.refresh_on is not None and self._token.refresh_on <= now)
```

Comment:
is there a need to account for jitter like the line below or is it okay to strictly refresh once that refresh period has gone?

Reply:
For refresh on, I think it's fine to adhere to it as-is without the buffer, so as not to refresh before the specified time. This keeps behavior consistent with .NET.